open_alchemy/cli.py

```diff
@@ -0,0 +1,105 @@
+"""Define the CLI module."""
+import argparse
+import logging
+import pathlib
+
+from open_alchemy import PackageFormat
+from open_alchemy import build_json
+from open_alchemy import build_yaml
+from open_alchemy import exceptions
+
+# Configure the logger.
+logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+# Define constants.
+VALID_EXTENSIONS = [".json", ".yml", ".yaml"]
+
+
+def main() -> None:
+    """Define the CLI entrypoint."""
+    args = build_application_parser()
+    try:
+        args.func(args)
+    except exceptions.BaseError as exc:
+        logging.error("Cannot perform the operation: %s.", exc)
+
+
+def validate_specfile(specfile: pathlib.Path) -> None:
+    """
+    Ensure the specfile is valid.
+
+    The specification file must:
+        * exist
+        * have a valid extension (.json, .yml, or .yaml)
+
+    Args:
+        specfile: Path object representing the specification file.
+    """
+    # Ensure the file exists.
+    if not specfile.exists():
+        raise exceptions.BaseError(f"the specfile '{specfile}' cannot be found")
```

Comment:
My apologies, I wasn't specific enough. I wouldn't expect `BaseError` to ever get raised, I would expect a more specific error to be raised that inherits from `BaseError`. `BaseError` is designed to go into a `try ... except ...` block to catch any OpenAlchemy error. I don't think there is a good error at the moment, perhaps a `CliError` can be defined here?